### PR TITLE
chore (refs DPLAN-11651): Add missing Hooks/Selectors for statements …

### PR DESCRIPTION
**Pull request by ahmad-demos**
### Ticket
https://demoseurope.youtrack.cloud/issue/DPLAN-11651

- Add missing Hooks/Selectors for statements list.

### CODE REVIEW

File: client/js/components/statement/assessmentTable/DpTable.vue [hunk 1]
```diff
@@ -170,11 +170,12 @@
...
       <template
         v-else
-        v-for="statement in statements">
+        v-for="(statement, key, index) in statements">
         <dp-assessment-table-card
           :csrf-token="csrfToken"
+          :data-cy="`statementCard:index:${index}`"
           :ref="'itemdisplay_' + statement.id"
-          :key="`statement:${statement.id}`"
+          :key="`statement:${key}`"
```
Comment (salisdemos) [inline client/js/components/statement/assessmentTable/DpTable.vue:178]: why this change?

**Review by salisdemos — CHANGES_REQUESTED** [review]

File: client/js/components/statement/assessmentTable/DpTable.vue (hunk 1 above)
Comment (ahmad-demos) [reply client/js/components/statement/assessmentTable/DpTable.vue:178]: When I added only index to v-for like `v-for="(statement, index) in statements"`
then data-cy resault: `data-cy="statementCard:index:fa0218f0-0175-4ead-be0f-e3dbffc6e27d"`

Comment (ahmad-demos) [reply client/js/components/statement/assessmentTable/DpTable.vue]: Done in https://github.com/demos-europe/demosplan-core/pull/3163/commits/8550398eebb5cc449d751c28bf462ff25a872dfe

**Review by salisdemos — APPROVED** [review]
